File: 18term2/COMP9041/poj/q18.pl
#!/usr/bin/perl -w
while($line = <>){
    my @nums = split /\s+/, $line;
    for my $i (@nums){
        $total_nums{$i} = 0 if($i =~ /\d+/);
    }
}
$min = (sort keys %total_nums)[0];
$max = (sort {$b <=> $a} keys %total_nums)[0];
while($min <= $max){
    if(!defined $total_nums{$min}){
        print "$min\n";
        last;
    }
    $min ++;
}

File: 18term2/COMP9041/poj/q18.py
#!/usr/bin/python3
import sys

nums_str = ''.join(sys.stdin).replace('\n', ' ')
nums_list = nums_str.split(' ')
nums_list = [x for x in nums_list if x != '']
nums_list = [int(x) for x in nums_list]
nums_list.sort()

for i in range(0, len(nums_list) - 1):
	if nums_list[i] + 1 != nums_list[i + 1]:
		print(nums_list[i] + 1)
		break
